test/system/table_test.rb

```diff
@@ -0,0 +1,132 @@
+require "application_system_test_case"
+
+class TableTest < ApplicationSystemTestCase
+  setup do
+    visit edit_post_path(posts(:empty))
+    wait_for_editor
+  end
+
+  test "adding a table" do
+    find_editor.toggle_command("insertTable")
+    sleep 0.1
+
+    html = find_editor.value
+    assert_match(/<table>/, html)
+    assert_table_structure(3, 3)
+  end
+
+  test "writing in table fields" do
+    find_editor.toggle_command("insertTable")
+    sleep 0.1
+
+    find_editor.send "Test Cell"
+    sleep 0.1
+
+    html = find_editor.value
+    assert_match(/Test Cell/, html)
+    assert_match(/<table>.*?Test Cell.*?<\/table>/m, html)
```

Comment:
Rather than manually sleeping, had you tried relying on Capybara's synchronization mechanisms that are built-in to its assertions?

```suggestion
    find_editor.toggle_command("insertTable")
    
    within find_editor do
      assert_selector :table

      find_editor.send "Test Cell"
    
      assert_selector :table, text: "Test Cell"
    end
```